src/xrGame/ActorCondition.cpp

```diff
@@ -643,12 +643,9 @@ void CActorCondition::DisableBoostParameters(const SBooster& B)
 }
 void CActorCondition::WoundForEach(const luabind::functor<bool>& funct)
 {
-    auto const& cur_wounds = wounds();
-    CEntityCondition::WOUND_VECTOR::const_iterator it = wounds().begin();
-    CEntityCondition::WOUND_VECTOR::const_iterator it_e = wounds().end();
-    for (; it != it_e; ++it)
+    for (auto it = wounds().begin(); it != wounds().end(); ++it)
     {
-        if (funct(it) == true)
+        if (funct(it))
```

Comment:
This functor accepts bool, but we are sending an iterator to it.
Something is wrong...

Also, keeping `end()` as a variable is still better, because `wounds().end()` will be called each time.